src/CSET/operators/regrid.py

```diff
@@ -0,0 +1,168 @@
+    y_coords = [coord for coord in coord_names if coord in Y_COORD_NAMES]
+    if len(y_coords) != 1:
+        raise ValueError("Could not identify a unique y-coordinate in cube")
+    y_coord = incube.coord(y_coords[0])
+
+    # List of supported grids - check if it is compatible
+    supported_grids = [iris.coord_systems.GeogCS]
+    if isinstance(incube.coord(x_coord).coord_system, supported_grids):
+        raise NotImplementedError(
+            f"Does not currently support {incube.coord(x_coord).coord_system} regrid method"
+        )
+    elif isinstance(incube.coord(y_coord).coord_system, supported_grids):
+        raise NotImplementedError(
+            f"Does not currently support {incube.coord(y_coord).coord_system} regrid method"
+        )
+
+    if method == "Linear":
+        return incube.regrid(target, iris.analysis.Linear())
+    else:
+        raise NotImplementedError(f"Does not currently support {method} regrid method")
+
+
+def regrid_onto_xyspacing(
+    incube: iris.cube.Cube, xspacing: int, yspacing: int, method: str, **kwargs
+) -> iris.cube.Cube:
+    """Regrid cube onto a set x,y spacing.
+
+    Regrid cube using specified x,y spacing, which is performed linearly.
+
+    Parameters
+    ----------
+    incube: Cube
+        An iris cube of the data to regrid. As a minimum, it needs to be 2D with a latitude,
+        longitude coordinates.
+    xspacing: integer
+        Spacing of points in longitude direction (could be degrees, meters etc.)
+    yspacing: integer
+        Spacing of points in latitude direction (could be degrees, meters etc.)
+    method: str
+        Method used to regrid onto, etc. Linear will use iris.analysis.Linear()
+
+    Returns
+    -------
+    cube_rgd: Cube
+        An iris cube of the data that has been regridded.
+
+    Raises
+    ------
+    ValueError
+        If a unique x/y coordinate cannot be found
+    NotImplementedError
+        If the cubes grid, or the method for regridding, is not yet supported.
+    """
+    # Usual names for spatial coordinates
+    X_COORD_NAMES = ["longitude", "grid_longitude", "projection_x_coordinate", "x"]
+    Y_COORD_NAMES = ["latitude", "grid_latitude", "projection_y_coordinate", "y"]
+
+    # Get a list of coordinate names for the cube
+    coord_names = [coord.name() for coord in incube.coords()]
+
+    # Check which x-coordinate we have, if any
+    x_coords = [coord for coord in coord_names if coord in X_COORD_NAMES]
+    if len(x_coords) != 1:
+        raise ValueError("Could not identify a unique x-coordinate in cube")
+    x_coord = incube.coord(x_coords[0])
+
+    # Check which y-coordinate we have, if any
+    y_coords = [coord for coord in coord_names if coord in Y_COORD_NAMES]
+    if len(y_coords) != 1:
+        raise ValueError("Could not identify a unique y-coordinate in cube")
+    y_coord = incube.coord(y_coords[0])
+
+    # List of supported grids - check if it is compatible
+    supported_grids = [iris.coord_systems.GeogCS]
```

Comment:
```suggestion
    supported_grids = (iris.coord_systems.GeogCS,)
```
isinstance needs it to be a tuple rather than a list. I presume it is due to the immutability?